powertools-e2e-tests/src/test/java/software/amazon/lambda/powertools/BatchE2ET.java

```diff
@@ -0,0 +1,279 @@
+                    .build());
+        }
+    }
+
+    @Test
+    public void sqsBatchProcessingSucceeds() throws InterruptedException {
+        List<SendMessageBatchRequestEntry> entries = testProducts.stream()
+                .map(p -> {
+                    try {
+                        return SendMessageBatchRequestEntry.builder()
+                                .id(p.getName())
+                                .messageBody(objectMapper.writeValueAsString(p))
+                                .build();
+                    } catch (JsonProcessingException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(Collectors.toList());
+
+        // WHEN
+        sqsClient.sendMessageBatch(SendMessageBatchRequest.builder()
+                .entries(entries)
+                .queueUrl(queueUrl)
+                .build());
+        Thread.sleep(30000); // wait for function to be executed
+
+        // THEN
+        ScanResponse items = ddbClient.scan(ScanRequest.builder()
+                .tableName(outputTable)
+                .build());
+        validateAllItemsHandled(items);
+    }
+
+    @Test
+    public void kinesisBatchProcessingSucceeds() throws InterruptedException {
+        List<PutRecordsRequestEntry> entries = testProducts.stream()
+                .map(p -> {
+                    try {
+                        return PutRecordsRequestEntry.builder()
+                                .partitionKey("1")
+                                .data(SdkBytes.fromUtf8String(objectMapper.writeValueAsString(p)))
+                                .build();
+                    } catch (JsonProcessingException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(Collectors.toList());
+
+        // WHEN
+        PutRecordsResponse result = kinesisClient.putRecords(PutRecordsRequest.builder()
+                .streamName(kinesisStreamName)
+                .records(entries)
+                .build());
+        Thread.sleep(30000); // wait for function to be executed
+
+        // THEN
+        ScanResponse items = ddbClient.scan(ScanRequest.builder()
+                .tableName(outputTable)
+                .build());
+        validateAllItemsHandled(items);
+    }
+
+    @Test
+    public void ddbStreamsBatchProcessingSucceeds() throws InterruptedException {
+        // GIVEN
+        String theId = "my-test-id";
+
+        // WHEN
+        ddbClient.putItem(PutItemRequest.builder()
+                .tableName(ddbStreamsTestTable)
+                .item(new HashMap<String, AttributeValue>() {
+                    {
+                        put("id", AttributeValue.builder()
+                                .s(theId)
+                                .build());
+                    }
+                })
+                .build());
+        Thread.sleep(90000); // wait for function to be executed
```

Comment:
this one needs more time ?

Reply:
Yeah, quite a bit more, it seems. I've got the batch timeout down on 1s and the batch size set appropriately in Infrastructure, but this seems to be about the comfortable wait.